client/my-sites/checkout/composite-checkout/record-analytics.js

```diff
@@ -104,6 +116,28 @@ export default function createAnalyticsEventHandler( reduxDispatch ) {
 					} )
 				);
 
+			case 'PAYMENT_METHOD_SELECT': {
+				reduxDispatch( logStashEventAction( 'payment_method_select', String( action.payload ) ) );
+
+				// Need to convert to the slug format used in old checkout so events are comparable
```

Comment:
This is fine but you could also add a new translate function alongside `translateCheckoutPaymentMethodToWpcomPaymentMethod` to extract this logic.